modules/hawkular-wildfly-agent-download/src/main/java/org/hawkular/component/wildflymonitor/WildFlyAgentServlet.java

```diff
@@ -86,37 +103,122 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
-        int limit = getDownloadLimit();
-        if (limit <= 0) {
-            sendErrorDownloadsDisabled(resp);
-            return;
-        } else if (limit < numActiveDownloads.get()) {
-            sendErrorTooManyDownloads(resp);
-            return;
+    private void downloadAgentModule(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+        try {
+            File agentModuleZip = getAgentModuleDownloadFile();
+            resp.setContentType("application/octet-stream");
+            resp.setHeader("Content-Disposition", "attachment; filename=" + agentModuleZip.getName());
+            resp.setContentLength((int) agentModuleZip.length());
+            resp.setDateHeader("Last-Modified", agentModuleZip.lastModified());
+            try (FileInputStream agentModuleZipStream = new FileInputStream(agentModuleZip)) {
+                copy(agentModuleZipStream, resp.getOutputStream());
+            }
+        } catch (Throwable t) {
+            String clientAddr = getClientAddress(req);
+            log("Failed to stream file to remote client [" + clientAddr + "]", t);
+            disableBrowserCache(resp);
+            resp.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Failed to stream file");
         }
+    }
 
+    private void downloadAgentInstaller(HttpServletRequest req, HttpServletResponse resp) throws IOException {
         try {
-            File agentJar = getAgentDownloadFile();
+            File agentInstallerJar = getAgentInstallerDownloadFile();
             resp.setContentType("application/octet-stream");
-            resp.setHeader("Content-Disposition", "attachment; filename=" + agentJar.getName());
-            resp.setContentLength((int) agentJar.length());
-            resp.setDateHeader("Last-Modified", agentJar.lastModified());
-            try (FileInputStream agentJarStream = new FileInputStream(agentJar)) {
-                copy(agentJarStream, resp.getOutputStream());
+            resp.setHeader("Content-Disposition", "attachment; filename=" + agentInstallerJar.getName());
+            resp.setContentLength((int) agentInstallerJar.length());
+            resp.setDateHeader("Last-Modified", agentInstallerJar.lastModified());
+
+            ZipFile agentInstallerZip = new ZipFile(agentInstallerJar);
+            ZipOutputStream zos = new ZipOutputStream(resp.getOutputStream(), StandardCharsets.UTF_8);
+            final String wildflyHome = getValueFromQueryParam(req, AGENT_INSTALLER_PROPERTY_WILDFLY_HOME, null);
+            final String hawkularServer = getValueFromQueryParam(req, AGENT_INSTALLER_PROPERTY_HAWKULAR_SERVER,
+                    "http://localhost:8080");
+            String defaultModuleDist = hawkularServer.startsWith("http:") ? hawkularServer +
+                    "/hawkular/wildfly-agent/download" : "http://localhost:8080/wildfly-agent/download";
```

Comment:
I also think you meant for the else clause in the ? : statement to be an https URL (schema=https with port 8443, with /hawkular/ in the path)